architecture/core_concepts/pods_and_services.adoc

```diff
@@ -507,6 +507,104 @@ incoming service connections between the endpoint pods. It also requires that
+If your application does not need load balancing or single-service IP, you can create a headless service. When you create a headless service, no load-balancing or proxying is done and no cluster IP is allocated for this service. For such services DNS is automatically configured depending on whether the service has selectors defined or not.
+
+*Services with selectors*: For headless services that define selectors, the endpoints controller creates `Endpoints` records in the API, and modifies the DNS configuration to return `A` records (addresses) that point directly to the pods backing the service.
+
+*Services without selectors*: For headless services that do not define selectors, the endpoints controller does not create `Endpoints` records. However, the DNS system looks for and configures either:
+
+* `CNAME` records for `ExternalName` type services. Or,
+* `A` records for any endpoints that share a name with the service, for all other types.
+
+[[headless-service-creation]]
+==== Creating a headless service
+Creating a headless service is similar to the creation of a standard service, but it does not declare the `ClusterIP`. To create a headless service, add `clusterIP: None` in the service YAML definition.
+
+For example, for a group of pods that you want to be a part of the same cluster/service.
+
+.List of pods
+[source, bash]
+----
+$ oc get pods
+NAME                        READY     STATUS      RESTARTS   AGE
+frontend-1-287hw            1/1       Running     0          7m
+frontend-1-68km5            1/1       Running     0          7m
+----
+
+You can define the headless service as:
+
+.Headless service definition
+[source, yaml]
+----
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app: ruby-helloworld-sample
+    template: application-template-stibuild
+  name: frontend-headless <1>
+spec:
+  clusterIP: None <2>
+  ports:
+  - name: web
+    port: 5432
+    protocol: TCP
+    targetPort: 8080
+  selector:
+    name: frontend <3>
+  sessionAffinity: None
+  type: ClusterIP
+status:
+  loadBalancer: {}
+----
+
+<1> Name of the headless servicce.
+<2> Setting `clusterIP` variable to `None` declares a headless service.
+<3> Selects all pods that have `frontend` label.
+
+Also, headless service do not have any IP address of its own.
+
+[source, bash]
+----
+$ oc get svc
+NAME                TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)    AGE
+frontend            ClusterIP   172.30.232.77    <none>        5432/TCP   12m
+frontend-headless   ClusterIP   None             <none>        5432/TCP   10m
+----
+
+[[headless-service-endpoint-discovery]]
+==== Endpoint discovery by using a headless service
+The benefit of using a headless service is that you can discover pod's IP address directly. Standard services act as load balancer or proxy, and give access to the workload object using the service name. Whereas in case of headless service, the service name resolves to the set of IP address of the pods grouped by the service.
```

Comment:
- "discover pod's" should be "discover a pod's"
- remove comma after proxy
- "using" should be "by using"
- I'd replace the last sentence with "With headless services, the service name resolves to the set of IP addresses of the pods that are grouped by the service."